var/spack/repos/builtin/packages/meep/package.py

```diff
@@ -39,18 +42,33 @@ class Meep(AutotoolsPackage):
     depends_on('lapack',      when='+lapack')
     depends_on('harminv',     when='+harminv')
     depends_on('guile',       when='+guile')
-    depends_on('libctl@3.2:', when='+libctl')
+    # according to the documentation meep need still depends on libctl @4.0: but
+    # according to the configure it is @4.2:
+    # https://github.com/NanoComp/meep/commit/97f4df0453417ff5b3876524845cc80b79f835e7
+    depends_on('libctl@4.2:', when='@1.6.2:')
+    depends_on('libctl@3.2:', when='@:1.6.1')
```

Comment:
The linked commit says it is present in tags since 1.8.0, and [previously](https://github.com/NanoComp/meep/commit/54d3abe631bea54f6f39262b432faf811bac44ae) it was libctl 4.1 from 1.5.0 onwards, and 4.0 since [1.4.0](https://github.com/NanoComp/meep/commit/2e1a2072ddd4c15d1b3faab9c03aaec4031a9502).

Also, I don't see any tags on GitHub for releases 1.6.1 and 1.6.2, are those a typo or where they published somewhere else?

I'm not sure if you tried all these old builds, but maybe we could reflect that in the dependencies?
```suggestion
    # according to the documentation meep need depends on libctl @4.0: but
    # according to the configure it is @4.2:
    # https://github.com/NanoComp/meep/commit/97f4df0453417ff5b3876524845cc80b79f835e7
    depends_on('libctl@4.2:', when='@1.8.0:')
    depends_on('libctl@4.1:', when='@1.5.0:1.7.0')
    depends_on('libctl@4.0:', when='@1.4.0:1.4.3')
    depends_on('libctl@3.2:', when='@:1.3')
```

Reply:
Additionally I just noticed that `libctl` is a variant, shouldn't these be conditional on enabling the variant?

Reply:
The variant is removed since the variant should be `python` and `scheme`